source/base/patterns.cc

```diff
@@ -780,15 +780,30 @@ namespace Patterns
       {
         unsigned int min_elements = 0, max_elements = 0;
 
-        std::istringstream is(description);
-        is.ignore(strlen(description_init) + strlen(" of <"));
+        std::string::const_iterator it =
+          description.begin() + strlen(description_init) + strlen(" of <");
 
-        std::string str;
-        std::getline(is, str, '>');
+        int  n_open_angular = 1;
+        auto tmp_it         = it - 1;
+        while (n_open_angular > 0)
+          {
+            tmp_it = std::find_if(tmp_it + 1, description.end(), [](char c) {
+              return (c == '>' || c == '<');
+            });
+            Assert(tmp_it != description.end(),
+                   ExcMessage("Couldn't find a closing '>' in description!"));
+            if (*tmp_it == '<')
+              ++n_open_angular;
+            else
+              --n_open_angular;
```

Comment:
A perpetual problem in parsing any kind of file is finding a matching closing parenthesis/token. I have no idea why the heck that has not ever been addressed by a standard function.